### PR TITLE
Add gcloudMode option to deploy

### LINKED ISSUE #354: Allow Specificying Beta when deploying
To use this functionality requires a `gcloud beta app deploy` command. 

Enable users to specifying the `beta` GAE environment with this plugin:
https://cloud.google.com/appengine/docs/standard/java/connecting-vpc

Thanks!

### PR DESCRIPTION
**Pull request by loosebazooka**
fixed #354 

### usage

```
appengine {
  deploy {
    ...
    gcloudMode = "beta"
  }
}
```

### CODE REVIEW

**Comment by loosebazooka** [comment]
Although I am wondering if I should have put this in the `appengine {}` configuration block directly?

**Comment by loosebazooka** [comment]
actually it looks like the `beta` component is installed automatically if you just use `beta` on the command line in `gcloud beta app deploy`. Gonna remove the component install section.

**Comment by chanseokoh** [comment]
Really? I tried `gcloud alpha survey` on the command line the other day and it prompted to install the alpha component. But maybe it doesn't prompt with `--quiet` and that is why it works in the plugin?

**Comment by loosebazooka** [comment]
I think beta and alpha are different?

**Comment by loosebazooka** [comment]
Anyway, I reworked the whole thing around that assumption. Will confirm with the cloud SDK team when I Can.

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java [hunk 1]
```diff
@@ -25,26 +25,35 @@
...
 import org.gradle.api.DefaultTask;
 import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Internal;
 import org.gradle.api.tasks.TaskAction;
 
 public class DownloadCloudSdkTask extends DefaultTask {
 
   private ManagedCloudSdk managedCloudSdk;
-  private boolean requiresAppEngineJava;
+  private List<SdkComponent> components = new ArrayList<>();
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java]: Can this be `final`?

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java [hunk 2]
```diff
@@ -25,26 +25,35 @@
...
-  public void requiresAppEngineJava(boolean requiresAppEngineJava) {
-    this.requiresAppEngineJava = requiresAppEngineJava;
+  public void requiresComponent(SdkComponent component) {
+    components.add(component);
+  }
+
+  @VisibleForTesting
+  @Internal
+  List<SdkComponent> getComponents() {
+    return components;
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java]: Return a copy?

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/AppEngineCorePluginConfiguration.java [hunk 3]
```diff
@@ -126,7 +127,11 @@ private void createDownloadCloudSdkTask() {
             downloadCloudSdkTask -> {
               downloadCloudSdkTask.setGroup(taskGroup);
               downloadCloudSdkTask.setDescription("Download the Cloud SDK");
-              downloadCloudSdkTask.requiresAppEngineJava(requiresAppEngineJava);
+
+              // make sure we download our required components
+              if (requiresAppEngineJava) {
+                downloadCloudSdkTask.requiresComponent(SdkComponent.APP_ENGINE_JAVA);
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/gradle/appengine/core/AppEngineCorePluginConfiguration.java]: There's no prod code that requests to install `SdkComponent.BETA`. Is the API change necessary?

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployExtension.java [hunk 4]
```diff
@@ -54,8 +55,10 @@ void setDeployTargetResolver(DeployTargetResolver deployTargetResolver) {
   DeployConfiguration toDeployConfiguration(List<Path> deployables) {
     String processedProjectId = deployTargetResolver.getProject(projectId);
     String processedVersion = deployTargetResolver.getVersion(version);
+
     return DeployConfiguration.builder(deployables)
         .bucket(bucket)
+        .gcloudMode(gcloudMode)
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployExtension.java]: Safe to pass null or an empty string?

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/AppEngineCorePluginConfiguration.java (hunk 3 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/appengine/core/AppEngineCorePluginConfiguration.java]: Yeah, I made these changes earlier to handle pre-isntalling BETA but then I liked them so I left it in.

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployExtension.java (hunk 4 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/appengine/core/DeployExtension.java]: yes, all our config options are safe to NULL

**Pull request merged**